game-core/src/main/java/games/strategy/triplea/settings/SelectionComponentFactory.java

```diff
@@ -724,14 +727,28 @@ public void reset() {
     };
   }
 
+  /**
+   * Creates UI Controls to fetch a login token and safely store it in the ClientSettings.
+   *
+   * @param uidSetting The uid setting used to potentially revoke old tokens
+   * @param usernameSetting The setting that stores the username being displayed to the user
+   * @param tokenSetting The setting that stores the actual token.
+   * @return A SelectionComponent that allows users to easily store their token.
+   */
   static SelectionComponent<JComponent> forumPosterSettings(
-      final ClientSetting<char[]> usernameSetting, final ClientSetting<char[]> passwordSetting) {
+      final String forumUrl,
+      final ClientSetting<Integer> uidSetting,
+      final ClientSetting<char[]> usernameSetting,
+      final ClientSetting<char[]> tokenSetting) {
     return new SelectionComponent<>() {
 
       private final JTextField usernameField =
-          new JTextField(credentialToString(usernameSetting::getValue), 20);
-      private final JPasswordField passwordField =
-          new JPasswordField(credentialToString(passwordSetting::getValue), 20);
+          new JTextFieldBuilder()
+              .columns(20)
+              .text(usernameSetting.getValue().map(String::new).orElse(""))
```

Comment:
I will properly clear any char arrays in the final form as well, even though I'm not sure if actually encrypting everything is that important now